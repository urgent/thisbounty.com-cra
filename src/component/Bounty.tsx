import React, { ReactChildren } from 'react'
import { Lifebar } from './Lifebar'
import { Moneybar } from './Moneybar'
import { Userbar } from './Userbar'
import styled from 'styled-components'

const Card = styled.div`
  display: grid;
  grid-template-columns: [image] 2fr [stats] 3fr;
  width: 320px;
  height: 192px;
  background-color: #212529;
  color: #fff;
  margin: 0 auto;
  box-shadow: 7px 7px 10px 0px rgb(0 0 0 / 50%);
  margin-bottom: 20px;
  border-radius: 10px;
  border: 4px solid rgba(50, 50, 50, 0.1);

  .shadow {
    position: absolute;
    width: 157px;
    height: 192px;
    box-shadow: inset 7px 7px 16px 3px rgb(0 0 0 / 67%);
    border-radius: 10px 0 0 10px;
  }

  > img {
    grid-column: image;
    width: 157px;
    height: 192px;
    border-radius: 10px 0 0 10px;
    &.center {
      background: black;
      display: flex;
    }
  }

  > .stats {
    grid-column: stats;
    display: grid;
    grid-template-rows: [title] 1fr [life] 1fr [money] 1fr [user] 1fr [programmer] 2fr [icons] 2fr;
    height: 184px;
    padding: 5px 5px 0px 0px;
  }

  .gutter {
    grid-row: gutter;
  }

  .lifebar {
    grid-row: life;
    margin-left: 5px;
  }

  .money {
    grid-row: money;
    margin-left: 5px;
  }

  .programmer {
    grid-row: programmer;
    vertical-align: middle;
    margin-left: 0px;
  }

  .user {
    grid-row: user;
  }

  .user img {
    margin: 0 2px;
  }

  .icon {
    display: flex;
    flex-direction: column-reverse;
    align-items: flex-end;
    margin-right: 4px;
  }
`

const TitleBar = styled.div`
  margin-left: 5px;
  grid-row: title;
  display: grid;
  grid-template-columns: [header] 5fr [id] 1fr;

  * {
    margin: 4px 0 0 0;
  }

  h2 {
    grid-column: header;
    font-size: 0.8rem;
    text-align: left;
  }

  h3 {
    grid-column: id;
    font-size: 1.2rem;
    height: 1.2rem;
    text-align: center;
  }
`

type Attributes = {
  id: number
  title: string
  image: string
  imageAlt: string
  life: number
  lifeMax: number
  lifeEnhance: boolean
  money: number
  moneyMax: number
  users: number
  usersMax: number
  programmers: number
  servers: number
  sourceCode: number
  references: number
  tags: String[]
  tagLinks: String[]
  hurtLog?: String[]
  children?: ReactChildren
}

export function Bounty ({
  id,
  title,
  image,
  imageAlt,
  life,
  lifeMax,
  lifeEnhance,
  money,
  moneyMax,
  users,
  usersMax,
  programmers,
  servers,
  sourceCode,
  references,
  hurtLog,
  children
}: Attributes) {
  return (
    <Card className='bounty'>
      <div className='shadow' />
      <img src={image} alt={imageAlt} />
      <div className='stats'>
        <TitleBar>
          <h2>{title}</h2>
        </TitleBar>
        <Lifebar life={life} max={lifeMax} enhance={lifeEnhance} />
        <Moneybar money={money} max={moneyMax} />
        <Userbar users={users} max={usersMax} />
        {children}
      </div>
    </Card>
  )
}
