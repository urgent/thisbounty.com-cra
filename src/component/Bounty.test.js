import renderer from 'react-test-renderer';
import React from 'react'
import { Bounty } from './Bounty'

const bounty = {
    bounty_id: 21,
    title: 'Get a client',
    life: 2,
    lifeMax: 3,
    money: 0,
    moneyMax: 0,
    users: 1,
    programmers: 1,
    servers: 0,
    sourceCode: 0,
    references: 0,
    tags: [],
    tagLinks: [],
    hurtLog: ['26th January, no launch, dead by 26th February', 'Worked for free, need to vet people more'],
};

test('Bounty renders correctly', () => {
    expect(renderer
        .create(<Bounty {...bounty} />)
        .toJSON()
    ).toMatchSnapshot();
});