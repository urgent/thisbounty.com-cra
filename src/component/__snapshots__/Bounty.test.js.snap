// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bounty renders correctly 1`] = `
<div
  className="sc-pFZIQ jTyBEv bounty"
>
  <div
    className="shadow"
  />
  <img />
  <div
    className="stats"
  >
    <div
      className="sc-jrAGrp jpsMnP"
    >
      <h2>
        Get a client
      </h2>
      <a
        className="sc-bdfBwQ cteGcY"
        href="#21"
      >
        #
        21
      </a>
    </div>
    <ul
      className="sc-dlfnbm dVZoKL lifebar"
    >
      <li>
        <span>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-heart fa-w-16 sc-gsTCUz koPeyr heart"
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={
              Object {
                "color": "#F00",
                "stroke": "none",
              }
            }
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </li>
      <li>
        <span>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-heart fa-w-16 sc-gsTCUz koPeyr heart"
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={
              Object {
                "color": "#F00",
                "stroke": "none",
              }
            }
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </li>
      <li>
        <span>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-heart fa-w-16 sc-gsTCUz koPeyr heart"
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={
              Object {
                "color": "rgba(0,0,0,0)",
                "stroke": "#F00",
              }
            }
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </li>
    </ul>
    <svg
      className="sc-hKgILt kWdQcq"
      fill="none"
      height="37"
      version="1.1"
      viewBox="0 0 180 37"
      width="170"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <g
        clipPath="url(#clip0)"
      >
        <g
          filter="url(#filter0_d)"
        >
          <path
            d="M14 14hNaNc5.059-.106 4.081 8.083-.793 7H14v-7z"
            fill="url(#linearGradient4670)"
          />
          <path
            d="M14 13h-1v9h-0.5a4.5 4.5 0 000-9z"
            stroke="#fff"
            strokeWidth="2"
          />
        </g>
        <path
          d="M9.5 12.347L5.25 5h7.969z"
          fill="#fff"
          stroke="#fff"
          strokeLinejoin="round"
          strokeWidth="3"
        />
        <path
          d="M9.5 25c4.694 0 8.5-3.655 8.5-8.163 0-4.509-3.806-8.164-8.5-8.164S1 12.328 1 16.837C1 21.345 4.806 25 9.5 25z"
          fill="#fff"
        />
        <path
          d="M10.377 17.632a.81.81 0 00-.253-.605c-.167-.165-.45-.313-.848-.446-.555-.17-.96-.39-1.215-.66-.252-.273-.379-.62-.379-1.04 0-.426.124-.776.371-1.05.25-.273.593-.436 1.028-.488v-.856h.582v.86c.437.06.777.243 1.02.55.244.308.366.728.366 1.262h-.718c0-.367-.088-.659-.262-.875a.86.86 0 00-.707-.324c-.31 0-.547.08-.711.242-.164.16-.246.382-.246.668 0 .266.086.478.258.637.174.156.46.3.855.43.399.127.708.27.93.425.224.154.388.334.492.54.107.205.16.446.16.722 0 .44-.133.794-.398 1.063-.263.268-.633.427-1.11.476v.746h-.578v-.746c-.484-.044-.864-.216-1.14-.515-.274-.302-.41-.711-.41-1.227h.722c0 .362.095.64.285.836.19.195.46.293.809.293.341 0 .61-.082.805-.246.195-.164.292-.388.292-.672z"
          fill="#343a40"
        />
      </g>
      <defs
        id="defs935"
      >
        <filter
          colorInterpolationFilters="sRGB"
          filterUnits="userSpaceOnUse"
          height="19"
          id="filter0_d"
          width="176"
          x="8"
          y="12"
        >
          <feFlood
            floodOpacity="0"
            id="feFlood912"
            result="BackgroundImageFix"
          />
          <feColorMatrix
            id="feColorMatrix914"
            in="SourceAlpha"
            type="matrix"
            values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
          />
          <feOffset
            dy="4"
            id="feOffset916"
          />
          <feGaussianBlur
            id="feGaussianBlur918"
            stdDeviation="2"
          />
          <feColorMatrix
            id="feColorMatrix920"
            type="matrix"
            values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
          />
          <feBlend
            id="feBlend922"
            in2="BackgroundImageFix"
            mode="normal"
            result="effect1_dropShadow"
          />
          <feBlend
            id="feBlend924"
            in="SourceGraphic"
            in2="effect1_dropShadow"
            mode="normal"
            result="shape"
          />
        </filter>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="paint0_linear"
          x1="14"
          x2="178"
          y1="21"
          y2="21"
        >
          <stop
            id="stop927"
            stopColor="#3E8832"
          />
          <stop
            id="stop929"
            offset="1"
            stopColor="#2BC012"
          />
        </linearGradient>
        <clipPath
          id="clip0"
        >
          <path
            d="M0 0h188v37H0V0z"
            fill="#fff"
            id="path932"
          />
        </clipPath>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="linearGradient4670"
          x1="14"
          x2="178"
          xlinkHref="#paint0_linear"
          y1="21"
          y2="21"
        />
      </defs>
    </svg>
    <div
      className="sc-eCssSg ifsWGG"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 sc-jSgupP gDjMXc"
        data-icon="user"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <svg
        className="sc-gKsewC UDbIK sc-hKgILt kWdQcq"
        fill="none"
        height="37"
        id="svg131"
        version="1.1"
        viewBox="0 0 180 37"
        width="170"
      >
        <g
          clipPath="url(#clip0)"
          id="g121"
          transform="translate(-12,-13.088235)"
        >
          <g
            id="g119"
          >
            <path
              d="m 14,14 h NaN c 5.059,-0.106 4.081,8.083 -0.793,7 H 14 Z"
              fill="url(#paint1_linear)"
              id="path115"
              style={
                Object {
                  "fill": "url(#paint1_linear)",
                }
              }
            />
            <path
              d="m 14,13 h -1 v 9 h NaN a 4.5,4.5 0 0 0 0,-9 z"
              id="path117"
              stroke="#ffffff"
              strokeWidth="2"
            />
          </g>
        </g>
        <defs
          id="defs935"
        >
          <linearGradient
            gradientUnits="userSpaceOnUse"
            id="paint1_linear"
            x1="14"
            x2="178"
            y1="21"
            y2="21"
          >
            <stop
              id="stop927"
              stopColor="#0000FF"
            />
            <stop
              id="stop929"
              offset="1"
              stopColor="#3377FF"
            />
          </linearGradient>
          <clipPath
            id="clip0"
          >
            <path
              d="M0 0h188v37H0V0z"
              fill="#fff"
              id="path932"
            />
          </clipPath>
        </defs>
      </svg>
    </div>
    <div
      className="sc-kEjbxe bfGhmx"
    >
      <div>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-users-cog fa-w-20 sc-iBPRYJ drFabV"
          data-icon="users-cog"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M610.5 341.3c2.6-14.1 2.6-28.5 0-42.6l25.8-14.9c3-1.7 4.3-5.2 3.3-8.5-6.7-21.6-18.2-41.2-33.2-57.4-2.3-2.5-6-3.1-9-1.4l-25.8 14.9c-10.9-9.3-23.4-16.5-36.9-21.3v-29.8c0-3.4-2.4-6.4-5.7-7.1-22.3-5-45-4.8-66.2 0-3.3.7-5.7 3.7-5.7 7.1v29.8c-13.5 4.8-26 12-36.9 21.3l-25.8-14.9c-2.9-1.7-6.7-1.1-9 1.4-15 16.2-26.5 35.8-33.2 57.4-1 3.3.4 6.8 3.3 8.5l25.8 14.9c-2.6 14.1-2.6 28.5 0 42.6l-25.8 14.9c-3 1.7-4.3 5.2-3.3 8.5 6.7 21.6 18.2 41.1 33.2 57.4 2.3 2.5 6 3.1 9 1.4l25.8-14.9c10.9 9.3 23.4 16.5 36.9 21.3v29.8c0 3.4 2.4 6.4 5.7 7.1 22.3 5 45 4.8 66.2 0 3.3-.7 5.7-3.7 5.7-7.1v-29.8c13.5-4.8 26-12 36.9-21.3l25.8 14.9c2.9 1.7 6.7 1.1 9-1.4 15-16.2 26.5-35.8 33.2-57.4 1-3.3-.4-6.8-3.3-8.5l-25.8-14.9zM496 368.5c-26.8 0-48.5-21.8-48.5-48.5s21.8-48.5 48.5-48.5 48.5 21.8 48.5 48.5-21.7 48.5-48.5 48.5zM96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm224 32c1.9 0 3.7-.5 5.6-.6 8.3-21.7 20.5-42.1 36.3-59.2 7.4-8 17.9-12.6 28.9-12.6 6.9 0 13.7 1.8 19.6 5.3l7.9 4.6c.8-.5 1.6-.9 2.4-1.4 7-14.6 11.2-30.8 11.2-48 0-61.9-50.1-112-112-112S208 82.1 208 144c0 61.9 50.1 112 112 112zm105.2 194.5c-2.3-1.2-4.6-2.6-6.8-3.9-8.2 4.8-15.3 9.8-27.5 9.8-10.9 0-21.4-4.6-28.9-12.6-18.3-19.8-32.3-43.9-40.2-69.6-10.7-34.5 24.9-49.7 25.8-50.3-.1-2.6-.1-5.2 0-7.8l-7.9-4.6c-3.8-2.2-7-5-9.8-8.1-3.3.2-6.5.6-9.8.6-24.6 0-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h255.4c-3.7-6-6.2-12.8-6.2-20.3v-9.2zM173.1 274.6C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         x 
        1
      </div>
      <div>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-server fa-w-16 sc-iBPRYJ drFabV"
          data-icon="server"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M480 160H32c-17.673 0-32-14.327-32-32V64c0-17.673 14.327-32 32-32h448c17.673 0 32 14.327 32 32v64c0 17.673-14.327 32-32 32zm-48-88c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24zm-64 0c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24zm112 248H32c-17.673 0-32-14.327-32-32v-64c0-17.673 14.327-32 32-32h448c17.673 0 32 14.327 32 32v64c0 17.673-14.327 32-32 32zm-48-88c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24zm-64 0c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24zm112 248H32c-17.673 0-32-14.327-32-32v-64c0-17.673 14.327-32 32-32h448c17.673 0 32 14.327 32 32v64c0 17.673-14.327 32-32 32zm-48-88c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24zm-64 0c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         x 
        0
      </div>
      <div>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-file-code fa-w-12 sc-iBPRYJ drFabV"
          data-icon="file-code"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 121.941V128H256V0h6.059c6.365 0 12.47 2.529 16.971 7.029l97.941 97.941A24.005 24.005 0 0 1 384 121.941zM248 160c-13.2 0-24-10.8-24-24V0H24C10.745 0 0 10.745 0 24v464c0 13.255 10.745 24 24 24h336c13.255 0 24-10.745 24-24V160H248zM123.206 400.505a5.4 5.4 0 0 1-7.633.246l-64.866-60.812a5.4 5.4 0 0 1 0-7.879l64.866-60.812a5.4 5.4 0 0 1 7.633.246l19.579 20.885a5.4 5.4 0 0 1-.372 7.747L101.65 336l40.763 35.874a5.4 5.4 0 0 1 .372 7.747l-19.579 20.884zm51.295 50.479l-27.453-7.97a5.402 5.402 0 0 1-3.681-6.692l61.44-211.626a5.402 5.402 0 0 1 6.692-3.681l27.452 7.97a5.4 5.4 0 0 1 3.68 6.692l-61.44 211.626a5.397 5.397 0 0 1-6.69 3.681zm160.792-111.045l-64.866 60.812a5.4 5.4 0 0 1-7.633-.246l-19.58-20.885a5.4 5.4 0 0 1 .372-7.747L284.35 336l-40.763-35.874a5.4 5.4 0 0 1-.372-7.747l19.58-20.885a5.4 5.4 0 0 1 7.633-.246l64.866 60.812a5.4 5.4 0 0 1-.001 7.879z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         x 
        undefined
      </div>
      <div>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-book fa-w-14 sc-iBPRYJ drFabV"
          data-icon="book"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         x 
        undefined
      </div>
    </div>
    <div
      className="sc-iqHYGH jFYpCI"
    />
  </div>
</div>
`;
